tests/test_cors.py

```diff
@@ -0,0 +1,164 @@
+import json
+from urllib.parse import parse_qs, urlparse
+
+import pytest
+from django.contrib.auth import get_user_model
+from django.test import RequestFactory, TestCase
+from django.urls import reverse
+
+from oauth2_provider.models import get_application_model
+
+from . import presets
+from .utils import get_basic_auth_header
+
+
+Application = get_application_model()
+UserModel = get_user_model()
+
+CLEARTEXT_SECRET = "1234567890abcdefghijklmnopqrstuvwxyz"
+
+# CORS is allowed for https only
+CLIENT_URI = "https://example.org"
+
+CLIENT_URI_HTTP = "http://example.org"
+
+
+@pytest.mark.usefixtures("oauth2_settings")
+@pytest.mark.oauth2_settings(presets.DEFAULT_SCOPES_RW)
+class TestCors(TestCase):
+    """
+    Test that CORS headers can be managed by OAuthLib.
+    The objective is: http request 'Origin' header should be passed to OAuthLib
+    """
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.test_user = UserModel.objects.create_user("test_user", "test@example.com", "123456")
+        self.dev_user = UserModel.objects.create_user("dev_user", "dev@example.com", "123456")
+
+        self.oauth2_settings.ALLOWED_REDIRECT_URI_SCHEMES = ["https"]
+        self.oauth2_settings.PKCE_REQUIRED = False
+
+        self.application = Application.objects.create(
+            name="Test Application",
+            redirect_uris=CLIENT_URI,
+            user=self.dev_user,
+            client_type=Application.CLIENT_CONFIDENTIAL,
+            authorization_grant_type=Application.GRANT_AUTHORIZATION_CODE,
+            client_secret=CLEARTEXT_SECRET,
+            allowed_origins=CLIENT_URI,
+        )
+
+        self.oauth2_settings.ALLOWED_REDIRECT_URI_SCHEMES = ["https"]
+
+    def tearDown(self):
+        self.application.delete()
+        self.test_user.delete()
+        self.dev_user.delete()
+
+    def test_cors_header(self):
```

Comment:
The test name should be clear about the case you are testing something more like  `test_valid_origin_with_https`. The endpoint and Cors is already captures in the class name.

Reply:
Done